packit_service/worker/build/build_helper.py

```diff
@@ -375,9 +418,34 @@ def _report(
         The status reporting should be done through this method
         so we can extend it in subclasses easily.
         """
-        self.status_reporter.report(
-            description=description, state=state, url=url, check_names=check_names,
-        )
+        if isinstance(self.project, GitlabProject) and not self.is_reporting_allowed:
+            # "Packit-User" does not have access to project.
+            self.request_project_access()
+            self.report_status_by_comment(state, url, check_names)
+        else:
+            self.status_reporter.report(
+                description=description, state=state, url=url, check_names=check_names,
+            )
```

Comment:
What about moving the `report_status_by_comment` to `status_reporter` class?

Reply:
Yes, I have moved `report_status_by_comment ` to `StatusReporter` class.